source/fuzz/fuzzer_pass_add_equation_instructions.cpp

```diff
@@ -64,12 +64,76 @@ void FuzzerPassAddEquationInstructions::Apply() {
 
         // Try the opcodes for which we know how to make ids at random until
         // something works.
-        std::vector<SpvOp> candidate_opcodes = {SpvOpIAdd, SpvOpISub,
-                                                SpvOpLogicalNot, SpvOpSNegate};
+        std::vector<SpvOp> candidate_opcodes = {
+            SpvOpIAdd,        SpvOpISub,        SpvOpLogicalNot,
+            SpvOpSNegate,     SpvOpConvertUToF, SpvOpConvertSToF,
+            SpvOpConvertFToU, SpvOpConvertFToS};
         do {
           auto opcode =
               GetFuzzerContext()->RemoveAtRandomIndex(&candidate_opcodes);
           switch (opcode) {
+            case SpvOpConvertFToS:
+            case SpvOpConvertFToU:
+            case SpvOpConvertSToF:
+            case SpvOpConvertUToF: {
+              auto candidate_instructions =
+                  opcode == SpvOpConvertFToS || opcode == SpvOpConvertFToU
+                      ? GetFloatInstructions(available_instructions)
+                      : GetIntegerInstructions(available_instructions);
+
+              if (candidate_instructions.empty()) {
+                break;
+              }
+
+              const auto* operand =
+                  candidate_instructions[GetFuzzerContext()->RandomIndex(
+                      candidate_instructions)];
+
+              const auto* type =
+                  GetIRContext()->get_type_mgr()->GetType(operand->type_id());
+              assert(type && "Operand has invalid type");
+
+              // Make sure a result type exists in the module.
+              switch (opcode) {
+                case SpvOpConvertFToS:
+                case SpvOpConvertFToU: {
+                  // OpConvertFToU can only convert to unsigned integer type.
```

Comment:
Make the comment explicit that OpConvertFToS can convert to a result of either type.

(Seems weird to me that there's this asymmetry - aside from this PR, are we sure that's deliberate?)

Reply:
Yes, the spec puts this constraint only on the `OpConvertFToU` instruction. This behaviour is consistent with the validator too (module is invalid if the result type of `OpConvertFToU` is a signed integer).

Reply:
Fixed in e4b2f8c.